src/tools/PrizeTierController/EditPrizeTiersModal/EditPrizeTierHistoryForm.tsx

```diff
@@ -0,0 +1,292 @@
+import { ErrorMessage } from '@components/ErrorMessage'
+import { StyledInput } from '@components/Input'
+import { Label } from '@components/Label'
+import { Button, ButtonTheme, ButtonSize } from '@pooltogether/react-components'
+import { calculate, calculateNumberOfPrizesForTierIndex } from '@pooltogether/v4-utils-js'
+import { EditPrizeTierFormValues } from '@prizeTierController/interfaces'
+import { getLastNonZeroTier } from '@prizeTierController/utils/getLastNonZeroTier'
+import classNames from 'classnames'
+import { utils } from 'ethers'
+import { useState } from 'react'
+import {
+  FieldErrorsImpl,
+  useForm,
+  UseFormRegister,
+  UseFormWatch,
+  useFieldArray,
+  Control
+} from 'react-hook-form'
+
+export const EditPrizeTierHistoryForm = (props: {
+  onSubmit: (prizeTier: EditPrizeTierFormValues) => void
+  defaultValues?: Partial<EditPrizeTierFormValues>
+  decimals: number
+}) => {
+  const { onSubmit, defaultValues, decimals } = props
+  const {
+    handleSubmit,
+    register,
+    control,
+    watch,
+    setValue,
+    formState: { errors, isValid }
+  } = useForm<EditPrizeTierFormValues>({
+    mode: 'onChange',
+    defaultValues,
+    shouldUnregister: true
+  })
+
+  // Reactively calculating prize value based on bitrange and tier changes:
+  const bitRange = parseInt(watch('bitRangeSize'))
+  const tierValues = watch('tiers').map((item) => Number(item.value))
+  if (!!bitRange && tierValues.length > 0 && tierValues.every((value) => !Number.isNaN(value))) {
```

Comment:
Same as the other comment, this math will run a lot. When you're dealing with user input you're better off using something like `useDebouncedCallback` <https://antonioru.github.io/beautiful-react-hooks/#/useDebouncedCallback>